### ISSUE TITLE
Error preprocessing due to probe map parameters?

### ISSUE THREAD

**Issue by hana777** [MouseLand/Kilosort2#192]
Hi everybody,

After converting my data into a proper binary file, I managed to succesfully load the data in the launcher. However, upon selecting a new probe layout, I try to run preprocessing. This yields the following error: 

Error preprocessing! Unable to perform assignment because the left and right sides have a different number of elements.

The probe I'm using is an H2 from NeuroTech: 2x32 channels, seperated 250 um. Does anybody have tips on how to insert the correct information for a new probe, using the GUI? Thanks in advance.

**Comment by marius10p** [comment]
Have you tried to follow the [wiki](https://github.com/MouseLand/Kilosort2/wiki/1.-The-GUI)?

**Comment by hana777** [comment]
Yes. However, the probe I am using has 2 shanks, each with 32 channels. This is different from the example in the wiki and therefore I wonder whether the probe parameters I have filled in might somehow be incorrect and therefore cause a problem in preprocessing and spikesorting. I have also tried to run a spikesorting session using artificial data generated with the eMouse simulator. These data seem to be sorted well, therefore I suspect that the problem lies with the probe map parameters.

The output file containing the voltage data has 64 rows. The odd-numbered rows belong to shank 1, the even-numbered rows correspond to the channels on shank 2. The two shanks are approximately 800 um long and located 250 um apart. 

These are the parameters I have filled in for the probe in the GUI:
xcoordinates: [ones(32,1); 250*ones(32,1)] 
_(32 channels with x=0, 32 channels with x=250)_

ycoordinates: [1:(800/32):800, (1:(800/32):800)] 
_(channels 1-32 have ycoordinates linearly spaced between 0-800, the same is true for channels 33-64)_

shank index: [ones(32,1); 2*ones(32,1)] 
_(the first 32 elements in the xcoord and ycoord arrays are located on shank 1, the others on shank 2)_

Channel map (the rows in the data file corresponding to each recording channel): [[1:2:64], [2:2:64]] 
_(so rows 1,3,5,7,...63 in the data file correspond to recording sites on shank 1, rows 2,4,6,....64 contain the data of shank 2)._

As far as I am aware, the values of the probe parameters are correct and should not cause any difficulties. However, there might be a problem because they are not in the correct format (e.g. they should be in a row vector instead of a column vector or vice versa). 

Do you know whether this is true and if so, what the correct Matlab expressions should be?

**Comment by hana777** [comment]
Can somebody answer my question please?

**Comment by RynzzZ** [comment]
Hi.

I have used the H2 probe from CambridgeNeurotech with KS2. Normally you would expect to create probe map file faithfully reflecting the probe configuration, for example channel1 being the bottom one on one shank, channel32 being the top, same thing for channel 33 and 64. But the thing is that your channel map and site location should also correspond to the associated data file - the CHs.dat file - you imported into KS2 GUI. 

I'm not sure what software you use for data collection, but I use openEphys. So the CHs.dat file I imported in KS2 GUI is generated by combining all the CH(number).continuous file together, and which channel the number associated with the .continous file reflects is determined by my channel mapping file loaded in openEphys, which is then determined by how your probe, adaptor and headstage connect together. Eventually, the probe mapping file I loaded in KS2 GUI is not as simple as   xcoordinates: [ones(32,1); 250*ones(32,1)]; ycoordinates: [1:(800/32):800, (1:(800/32):800)]. For example, CH1.continuous file actually coorespond to channel 41 on my H2 probe, so the xcoordinates(1) and ycoordinates(1) should be the x and y for channel 41, instead of the bottom channel of the left shank.

Hopefully this might help.
